compiler/pash.py

```diff
@@ -67,29 +70,34 @@ def parse_args():
         prog_name = os.environ['PASH_FROM_SH']
     parser = argparse.ArgumentParser(prog_name)
     parser.add_argument("input", help="the script to be compiled and executed")
-    parser.add_argument("--preprocess_only", 
+    parser.add_argument("--preprocess_only",
                         help="only preprocess the input script and not execute it",
                         action="store_true")
-    parser.add_argument("--output_preprocessed", 
+    parser.add_argument("--output_preprocessed",
                         help=" output the preprocessed script",
                         action="store_true")
     config.add_common_arguments(parser)
     args = parser.parse_args()
     return args
 
 def preprocess(ast_objects, config):
-    ## This is ids for the remporary files that we will save the IRs in
+    ## This is ids for the temporary files that we will save the IRs in
     irFileGen = FileIdGen()
 
-    ## Preprocess ASTs by replacing AST regions with calls to PaSh's runtime. 
-    ## Then the runtime will do the compilation and optimization with additional 
+    ## Preprocess ASTs by replacing AST regions with calls to PaSh's runtime.
+    ## Then the runtime will do the compilation and optimization with additional
     ## information.
     preprocessed_asts = replace_ast_regions(ast_objects, irFileGen, config)
 
     return preprocessed_asts
 
-def execute_script(compiled_script_filename):
-    exec_obj = subprocess.run(["/usr/bin/env", "bash" ,compiled_script_filename])
+def execute_script(compiled_script_filename, debug_level):
+    new_env = os.environ.copy()
+    new_env["PASH_TMP_PREFIX"] = config.PASH_TMP_PREFIX
+    exec_obj = subprocess.run(["/usr/bin/env", "bash" ,compiled_script_filename], env=new_env)
+    ## Delete the temp directory when not debugging
+    if(debug_level == 0):
+        shutil.rmtree(config.PASH_TMP_PREFIX)
```

Comment:
appart from checking debug_level, it'd be good to make sure that PASH_TMP_PREFIX was not provided by the user, but created ourselves. It's not 

`PASH_TMP_DIR=~/ pash.sh` could have interesting consequences :)

Reply:
Thanks, very good point :) I modified it so that `pash.py` (the entrance for pash) always sets this itself (and only the runtime that doesn't do any deleting inherits it).

Reply:
I don't fully get the cycle of pash compile vs runtime.... but if it works fine... that's ok. 

some tests wouldn't hurt either :)

Reply:
At the moment the standard tests (in `compiler/test_evaluation_scripts.sh`) are OK witnesses of correctness. I tried to think about some test to check that directories are deleted correctly, but it would be very elaborate for what it checks I think.

So I guess that we could merge if you don't have any idea of a straightforward to do test to check that. Do you have any other property to check in mind?

Reply:
I am merging the PR and let's see if we want additional tests later!